### PR TITLE
Cephadm: created separate test modules for bootstrap and host operations

### PR DESCRIPTION
**Pull request by sunilkumarn417**
Signed-off-by: sunilkumarn417 <sunnagar@redhat.com>

- Added separate test modules for bootstrap and host operations.
- Moved out bootstrap verification from bootstrap module to test_bootstrap.

**Results:**
Sanity - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1614927395117/result.html 
Deployment - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1614931512053/result.html

### CODE REVIEW

File: tests/ceph_installer/test_cephadm.py [hunk 1]
```diff
@@ -1,95 +1,121 @@
...
+from ceph.ceph_admin.nfs import NFS
+from ceph.ceph_admin.node_exporter import NodeExporter
+from ceph.ceph_admin.osd import OSD
+from ceph.ceph_admin.prometheus import Prometheus
+from ceph.ceph_admin.rgw import RGW
+
+LOG = logging.getLogger()
+SERVICE_MAP = dict(
+    {
+        "alertmanager": AlertManager,
```
[the remark this answers is not in the document]
Comment (sunilkumarn417) [reply tests/ceph_installer/test_cephadm.py:29]: Done and only Crash was missing.

File: tests/cephadm/test_bootstrap.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        str_ = r"%s:\s(.+)" % string
+        return re.search(str_, db_log_part).group(1)
+
+    host = parse("URL")
+    user_ = parse("User")
+    passwd_ = parse("Password")
+
+    if user:
+        if user != user_:
+            raise BootStrapValidationFailure(f"user {user} did not match")
```
[the remark this answers is not in the document]
Comment (sunilkumarn417) [reply tests/cephadm/test_bootstrap.py:75]: Done

File: tests/cephadm/test_bootstrap.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    session = requests.Session()
+    session.headers = {
+        "accept": "application/vnd.ceph.api.v1.0+json",
+        "content-type": "application/json",
+    }
+
+    resp = session.post(url_, data=data, verify=False)
+    if not resp.ok:
+        raise BootStrapValidationFailure(
+            "Response: %s, Status code: %s" % (resp.text, resp.status_code)
```
[the remark this answers is not in the document]
Comment (sunilkumarn417) [reply tests/cephadm/test_bootstrap.py]: updated

File: tests/cephadm/test_bootstrap.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    Method to validate dashboard login using provided user/password
+    and Also validates user/passwd in bootstrap console output
+
+    Args:
+        cls: class object
+        out: bootstrap response log
+        user: dashboard user
+        password: dashbard password
+
+    Returns:
```
[the remark this answers is not in the document]
Comment (sunilkumarn417) [reply tests/cephadm/test_bootstrap.py]: Updated

File: tests/cephadm/test_bootstrap.py [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+    # Manage Ceph using ceph-admin orchestration
+    command = config.pop("command")
+    service = config.pop("service", "")
+
+    log.info("Executing %s %s" % (service, command))
+    instance = CephAdmin(cluster=ceph_cluster, **config)
+    method = fetch_method(instance, command)
+
+    if "shell" in command:
```
[the remark this answers is not in the document]
Comment (sunilkumarn417) [reply tests/cephadm/test_bootstrap.py:179]: Done

File: tests/cephadm/test_bootstrap.py [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+                    custom_image: true | false
+                    mon-ip: <node_name>
+                    mgr-id: <mgr_id>
+                    fsid: <id>
+    """
+    config = kw.get("config")
+    build = config.get("build", config.get("rhbuild"))
+    ceph_cluster.rhcs_version = build
+
+    if config.get("skip_setup") is True:
```
[the remark this answers is not in the document]
Comment (sunilkumarn417) [reply tests/cephadm/test_bootstrap.py]: Removed `skip_setup`.

**Review by psathyan — APPROVED** [review]

**Pull request merged**